crates/web5_cli/src/doctor.rs

```diff
@@ -0,0 +1,218 @@
+use std::collections::HashMap;
+
+use crate::CheckType;
+
+pub struct HealthCheckState {
+    cli_version: Option<Option<String>>,
+    dependencies: Option<HashMap<String, bool>>,
+    env_vars: Option<HashMap<String, bool>>,
+    connectivity: Option<bool>,
+    basic_functionality: Option<bool>,
+}
+
+impl HealthCheckState {
+    fn new() -> Self {
+        Self {
+            cli_version: None,
+            dependencies: None,
+            env_vars: None,
+            connectivity: None,
+            basic_functionality: None,
+        }
+    }
+}
+
+fn check_cli_version() -> Option<Option<String>> {
+    Some(Some(env!("CARGO_PKG_VERSION").to_string())) // This will return the binary version.
+}
```

Comment:
```suggestion
fn check_cli_version() -> Option<String> {
    Some(env!("CARGO_PKG_VERSION").to_string())
}
```

can we remove the double Option and Some

Reply:
We can but the first reason why I kept it over there is because let's say you want to run individual tests then in the print function , I needed a way to check if a user has asked for a certain check or not. 

Let me re-think this in some other possible way.